### PR TITLE
Implement ReportUser RPC, minor fixes around the reporting UI

### PR DESCRIPTION
**Pull request by zapu**
Adds an RPC for new user reporting API call in https://github.com/keybase/keybase/pull/4780

It works for both reporting from user profile or chat. When reporting from chat, we will always send over Conversation ID (even if we are not sending transcripts). This way server can associate particular report with a TLF.

TODO:
- [x] Disable transcript sending in UI as it's not implemented. Decide if we should do it via some kind of feature flag or just disable it for everyone.

### CODE REVIEW

**Comment by zapu** [comment]
cc @buoyad commented out transcripts + some other fixes around block modal

File: go/service/user.go [hunk 1]
```diff
@@ -737,10 +737,27 @@ func (h *UserHandler) ReportUser(ctx context.Context, arg keybase1.ReportUserArg
 		arg.Username, arg.IncludeTranscript, convIDStr),
 		func() error { return err })()
 
-	if arg.IncludeTranscript && arg.ConvID == nil {
-		return errors.New("invalid arguments: IncludeTranscript is true but ConvID == nil")
+	if arg.IncludeTranscript {
+		if arg.ConvID == nil {
+			return errors.New("invalid arguments: IncludeTranscript is true but ConvID == nil")
+		}
+		return errors.New("`IncludeTranscript` is not implemented")
```
Comment (heronhaye) [inline go/service/user.go]: so this is always going to be false for the release?

Comment (zapu) [reply go/service/user.go]: hopefully... this PR also comments the checkbox out in the GUI

Comment (heronhaye) [reply go/service/user.go]: maybe safer to log and ignore for now if it's not clear. so we don't end up with a release where blocking is impossible in some cases

Comment (zapu) [reply go/service/user.go]: good idea

**Pull request merged**